AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,126 @@
-
+    var movieURL: URL?
+    var movies: [AVPlayerLayer] = [AVPlayerLayer]()
+    
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
+    
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
+        
+        /*
+         Thi stuff is partially from https://github.com/martyav/AC3.2-AVPlayerKVO/blob/master/AVPlayerKVO/ViewController.swift but altered to fit the part
+         */
+        
+        guard let topSubLayer = self.videoContainerTop.layer.sublayers,
+            let bottomSubLayer = self.videoContainerBottom.layer.sublayers
+            else {
+                return
+        }
+        
+        for layer in topSubLayer {
+            layer.frame = self.videoContainerTop.bounds
+        }
+        
+        for layer in bottomSubLayer {
+            layer.frame = self.videoContainerBottom.bounds
+        }
+        
+        view.layoutIfNeeded()
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .photoLibrary
+        // recall that, by default, this array only contains static pix...
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
+        imagePickerController.delegate = self
+        self.present(imagePickerController, animated: true, completion: nil)
+    }
     
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+            self.movieURL = url
+            
+            let playerItem = AVPlayerItem(url: url)
+            let player = AVPlayer(playerItem: playerItem)
+            let playerLayer = AVPlayerLayer(player: player)
+            
+            // looking at http://stackoverflow.com/questions/29000251/swift-resize-an-avplayerlayer-to-match-the-bounds-of-its-parent-container
+            
+            movies.append(playerLayer)
+            
+            if self.videoContainerTop.layer.sublayers == nil && self.videoContainerBottom.layer.sublayers == nil {
+                self.videoContainerTop.layer.addSublayer(playerLayer)
+                playerLayer.frame = videoContainerTop.bounds
+                player.play()
+            } else if videoContainerBottom.layer.sublayers == nil {
+                self.videoContainerBottom.layer.addSublayer(playerLayer)
+                playerLayer.frame = videoContainerBottom.bounds
+                player.play()
+            } else {
+                let top = movies[0]
+                let bottom = movies[1]
+                
+                if top.player?.rate != 0 && bottom.player?.rate == 0 {
```

Comment:
Hi. I'm not sure what went wrong. I thought I had it working when I submitted. At any rate, I did manage to get more videos to play after working on the assessment on my branch, NextDay